src/overlays/actors/ovl_En_Attack_Niw/z_en_attack_niw.c

```diff
@@ -27,23 +36,372 @@ const ActorInit En_Attack_Niw_InitVars = {
+        func_809B5268(this, globalCtx, 2);
+    }
+}
+
+void EnAttackNiw_Update(Actor* thisx, GlobalContext* globalCtx) {
+    f32 tmpf1;
+    EnAttackNiw* this = THIS;
+    EnNiw* actor;
+    Player* player = PLAYER;
+    s32 pad;
+    Vec3f sp30;
+    GlobalContext* ctx2 = globalCtx;
+
+    this->unk_28C++;
+    if (this->unk_254 != 0) {
+        this->unk_254--;
+    }
+    if (this->unk_258 != 0) {
+        this->unk_258--;
+    }
+    if (this->unk_25A != 0) {
+        this->unk_25A--;
+    }
+    if (this->unk_25E != 0) {
+        this->unk_25E--;
+    }
+    if (this->unk_260 != 0) {
+        this->unk_260--;
+    }
+    if (this->unk_25C != 0) {
+        this->unk_25C--;
+    }
+    if (this->unk_262 != 0) {
+        this->unk_262--;
+    }
+
+    this->actor.shape.rot = this->actor.posRot.rot;
+    this->actor.shape.unk_10 = 15.0f;
+    this->actionFunc(this, ctx2);
+    func_8002E4B4(globalCtx, &this->actor, 20.0f, 20.0f, 60.0f, 0x1D);
+
+    if (this->actionFunc == func_809B5670) {
+        func_8002D97C(&this->actor);
+    } else {
+        Actor_MoveForward(&this->actor);
+    }
+
+    if (this->actor.groundY <= -32000.0f) {
+        Actor_Kill(&this->actor);
+        return;
+    }
+
+    if ((this->actor.bgCheckFlags & 0x20) && (this->actionFunc != func_809B5C18)) {
+        Math_Vec3f_Copy(&sp30, &this->actor.posRot.pos);
+        sp30.y += this->actor.waterY;
+        func_8002949C(globalCtx, &sp30, 0, 0, 0, 0x190);
+        this->unk_2DC = 0.0f;
+        this->actor.gravity = 0.0f;
+        this->unk_2E0 = 0.0f;
+        this->unk_2D0 = this->actor.posRot.rot.x - 5000.0f;
+        this->actionFunc = func_809B5C18;
+        return;
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Attack_Niw/func_809B59B0.s")
+    tmpf1 = 20.0f;
+    if (this->actor.xyzDistFromLinkSq < SQ(tmpf1)) {
+        actor = (EnNiw*)this->actor.attachedA;
+        if ((this->actor.attachedA->update != NULL) && (this->actor.attachedA != NULL) && (actor != NULL) &&
+            (actor->unk_26A == 0) && (player->invincibilityTimer == 0)) {
+            func_8002F6D4(globalCtx, &this->actor, 2.0f, this->actor.posRot.rot.y, 0.0f, 0x10);
+            actor->unk_26A = 0x46;
+        }
+    }
+    if (this->unk_25E == 0) {
+        this->unk_25E = 30;
+        Audio_PlayActorSound2(&this->actor, NA_SE_EV_CHICKEN_CRY_A);
+    }
+    if (this->unk_260 == 0) {
+        this->unk_260 = 7;
+        Audio_PlayActorSound2(&this->actor, NA_SE_EN_DEKU_WAKEUP);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Attack_Niw/func_809B5C18.s")
+s32 func_809B5F98(GlobalContext* globalCtx, s32 limbIndex, Gfx** dList, Vec3f* pos, Vec3s* rot, Actor* thisx) {
+    EnAttackNiw* this = THIS;
+    Vec3f sp0 = { 0.0f, 0.0f, 0.0f };
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Attack_Niw/EnAttackNiw_Update.s")
+    if (limbIndex == 13) {
+        rot->y += (s16)this->unk_2BC;
```

Comment:
Are these casts necessary? I can't imagine they would be

Reply:
Yep, required for matching.